spec/directives/use/load.hrx

```diff
@@ -0,0 +1,453 @@
+}
+
+<===>
+================================================================================
+<===> css_order/unrelated_branches/input.scss
+@use "left_midstream";
+@use "right_midstream";
+
+a {file: input}
+
+<===> css_order/unrelated_branches/left_midstream.scss
+@use "left_upstream";
+
+a {file: left midstream}
+
+<===> css_order/unrelated_branches/left_upstream.scss
+a {file: left upstream}
+
+<===> css_order/unrelated_branches/right_midstream.scss
+@use "right_upstream";
+
+a {file: right midstream}
+
+<===> css_order/unrelated_branches/right_upstream.scss
+a {file: right upstream}
+
+<===> css_order/unrelated_branches/output.css
+a {
+  file: left upstream;
+}
+
+a {
+  file: left midstream;
+}
+
+a {
+  file: right upstream;
+}
+
+a {
+  file: right midstream;
+}
+
+a {
+  file: input;
+}
+
+<===>
+================================================================================
+<===> import_order/use_into_use/input.scss
+@use "midstream";
+
+@import "input.css";
+
+<===> import_order/use_into_use/midstream.scss
+@use "upstream";
+
+@import "midstream.css";
+
+a {file: midstream}
+
+<===> import_order/use_into_use/upstream.scss
+@import "upstream.css";
+
+<===> import_order/use_into_use/output.css
+@import "upstream.css";
+@import "midstream.css";
+@import "input.css";
+
+a {
+  file: midstream;
+}
+
+<===>
+================================================================================
+<===> import_order/import_below_rule/input.scss
+@use "midstream";
+
+@import "input.css";
+
+<===> import_order/import_below_rule/midstream.scss
+@use "upstream";
+
+a {file: midstream}
+
+@import "midstream.css";
+
+<===> import_order/import_below_rule/upstream.scss
+@import "upstream.css";
+
+<===> import_order/import_below_rule/output.css
+@import "upstream.css";
+@import "midstream.css";
+@import "input.css";
+
+a {
+  file: midstream;
+}
+
+<===>
+================================================================================
+<===> import_order/comments_and_imports/input.scss
+/* input comment before use */
+@use "midstream";
+
+/* input comment before import */
+@import "input.css";
+
+/* input comment after import */
+
+<===> import_order/comments_and_imports/midstream.scss
+/* midstream comment before use */
+@use "upstream";
+
+/* midstream comment before first import */
+@import "midstream1.css";
+
+/* midstream comment before second import */
+@import "midstream2.css";
+
+/* midstream comment after imports */
+
+a {file: midstream}
+
+<===> import_order/comments_and_imports/upstream.scss
+/* upstream comment before import */
+@import "upstream.css";
+
+/* upstream comment after import */
+
+<===> import_order/comments_and_imports/output.css
+/* upstream comment before import */
+
+@import "upstream.css";
+
+/* midstream comment before use */
+
+/* midstream comment before first import */
+
+@import "midstream1.css";
+
+/* midstream comment before second import */
+
+@import "midstream2.css";
+
+/* input comment before use */
```

Comment:
Why do the "before use" comments appear after the contents of the `@use` file?

Reply:
It would be *very* complicated to keep track of which comments appeared above or between which `@use` rules in the source text and use that to interpolate them appropriately into the CSS, especially once you get into multiple `@use`s of the same module. The next best thing is just to keep the contents of an individual file together as much as possible (with the caveat that CSS `@import`s are split out).